python/tvm/script/parser.py

```diff
@@ -272,6 +272,13 @@ def parse_arg_list(self, func, node_call):
                 f"but it is {type(func).__name__}",
                 node_call.span,
             )
+
+        # for loop syntax sugar, check if starting 0 is omitted
+        # param_list[0] is the list of positional args which could include kw_args
+        # therefore the sum of len(args) and len(kw_args) is used here
+        if isinstance(func, ForScopeHandler) and len(args) + len(kw_args) < len(param_list[0]):
+            if args[0] != 0:
+                args.insert(0, 0)
```

Comment:
if possible I'd prefer implementing such logic in `ForScopeHandler` (or its subclasses like `Parallel`, `Vectorized`, etc.) to separate the details of these statements from the parser. see the example here https://github.com/apache/tvm/blob/main/python/tvm/script/tir/scope_handler.py#L591-L593

Reply:
done